q2_moshpit/dereplication/derep.py

```diff
@@ -257,22 +257,67 @@ def _generate_pa_table(
     return presence_absence
 
 
+def _get_representatives(mags, busco_results, bin_clusters):
+    """
+    This function identifies the representative bin for each cluster of bins.
+    If `busco_results` is provided, the selection is first based on
+    completeness, and in case of a tie, the longest bin is chosen. If
+    `busco_results` is not available, the longest bin is selected by default.
+
+    Args:
+        mags: A MultiMAGSequencesDirFmt object containing all bins.
+        busco_results: A DataFrame containing BUSCO results.
+        bin_clusters: A list of lists where each inner list contains the IDs
+                      of bins grouped by similarity.
+
+    Returns:
+        A list of representative bin IDs, one for each cluster.
+    """
+    bin_lengths = _get_bin_lengths(mags)
+
+    # Choose by BUSCO results
+    if busco_results is not None:
+        bin_completeness = busco_results['complete']
+
+        representative_bins = []
+        for bins in bin_clusters:
+            # Get bins with the highest completeness values in cluster
+            completest_bins = (completeness_values := bin_completeness[bins])[
+                completeness_values == completeness_values.max()].index
+
+            # If there's a tie, resolve by selecting the longest bin
+            if len(completest_bins) > 1:
+                lengths_of_best_bins = bin_lengths[completest_bins]
+                representative_bins.append(lengths_of_best_bins.idxmax())
+            else:
+                representative_bins.append(completest_bins[0])
```

Comment:
You could simplify this to:
```suggestion
    if busco_results is not None:
        bin_completeness = busco_results['complete']
    
        representative_bins = []
        for bins in bin_clusters:
            # Get bins with the highest completeness values in cluster
            completest_bins = bin_completeness[bins].idxmax()
    
            # If there's a tie, resolve by selecting the longest bin
            if len(completest_bins) > 1:
                representative_bins.append(bin_lengths[completest_bins].idxmax())
            else:
                representative_bins.append(completest_bins[0])
```

Reply:
This does not work as expected, because this will give me the first id of the bin with the highest value. But I need all ids with the highest value if there are multiple.